### PR TITLE
[Debugger] Fix showing the type of stream variables

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/29456 - Error when showing type/value of streams with type "stream<string[], io:Error>

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BStream.java [hunk 1]
```diff
@@ -21,38 +21,39 @@
...
         }
     }
+
+    /**
+     * Some additional processing is required, as `stringValue()` output of the runtime stream values contains some
+     * redundant information.
+     */
+    private static String processStringValue(String stringValue) {
+        if (stringValue.startsWith("stream <stream") && stringValue.endsWith(">>")) {
+            stringValue = stringValue.replaceFirst("stream <stream", "stream").replaceFirst(">>$", ">");
```
Comment (praveennadarajah) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BStream.java:52]: What are the possible `stringValue ` are we getting here?

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BStream.java [hunk 2]
```diff
@@ -21,38 +21,39 @@
...
+        }
+        if (stringValue.startsWith("stream<(") && stringValue.endsWith(")>")) {
+            stringValue = stringValue.replaceFirst("stream<\\(", "stream <").replaceFirst("\\)>$", ">");
```
Comment (praveennadarajah) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BStream.java:55]: What are the possible `stringValue` are we getting here?

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BStream.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BStream.java:52]: We are doing these modifications in the debugger side, due to some runtime conversions of the `stringValue` method.

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BStream.java (hunk 2 above)
Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BStream.java:55]: https://github.com/ballerina-platform/ballerina-lang/pull/30611#discussion_r633481005

**Review by praveennadarajah — DISMISSED** [review]

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/evaluation/ExpressionEvaluationTest.java [hunk 3]
```diff
@@ -159,7 +159,7 @@ public void variableReferenceEvaluationTest() throws BallerinaTestException {
         // table variable test
         debugTestRunner.assertExpression(context, TABLE_VAR, "table<Employee>[3]", "table");
         // stream variable test
-        debugTestRunner.assertExpression(context, STREAM_VAR, "stream<int>", "stream");
+        debugTestRunner.assertExpression(context, STREAM_VAR, "stream<int,error>", "stream");
```
Comment (Dilhasha) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/evaluation/ExpressionEvaluationTest.java:162]:
```suggestion
        debugTestRunner.assertExpression(context, STREAM_VAR, "stream<int, error>", "stream");
```

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java [hunk 4]
```diff
@@ -276,7 +276,7 @@ public void localVariableVisibilityTest() {
         debugTestRunner.assertVariable(localVariables, "tableVar", "table<Employee>[3]", "table");
 
         // stream variable visibility test
-        debugTestRunner.assertVariable(localVariables, "oddNumberStream", "stream<int>", "stream");
+        debugTestRunner.assertVariable(localVariables, "oddNumberStream", "stream<int,error>", "stream");
```
Comment (Dilhasha) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java:279]:
```suggestion
        debugTestRunner.assertVariable(localVariables, "oddNumberStream", "stream<int, error>", "stream");
```

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/evaluation/ExpressionEvaluationTest.java (hunk 3 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/evaluation/ExpressionEvaluationTest.java]: Addressed in e41f309

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java (hunk 4 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java]: Addressed in e41f309

**Review by praveennadarajah — APPROVED** [review]

**Pull request merged**